core/shell.py

```diff
@@ -287,6 +288,17 @@ def AddBlock(builtins, mem, mutable_opts, dir_stack, cmd_ev, shell_ex,
             hay_state, mem, cmd_ev)
 
 
+def AddMethods(methods):
+    # type: (Dict[int, Dict[str, vm._Callable]]) -> None
+    """Initialize methods table."""
+    methods[value_e.Str] = {
+        'startswith': func_misc.StartsWith(),
```

Comment:
Another thing we should think about is whether it should be `s->startsWith("a")`

JavaScript uses that spelling

It's kind annoying to have it inconsistent with Python, but there will probably be more functions later which we'll want to be consistent with

I think right now it's `eval_hay()` and `parse_hay()`, and those should probably be `evalHay()` and `parseHay()`

We don't have to decide now, it's probably better to make a pass through everything, and then make these decisions when we **write the docs**

Reply:
Yes, I originally went with `startsWith` but found that `spec/ysh-user-feedback.test.sh` used `startswith` instead. So I decided to follow the precedent (and avoid a breaking change, if that even matters right now.)

Eventually, I imagine that we'd want to create our own style guide. Sort of like Python's PEP8.